app/uk/gov/hmrc/contactadvisors/service/EmailService.scala

```diff
@@ -49,11 +47,9 @@ class EmailService @Inject()(emailConnector: EmailConnector, conf: Configuration
       }
       .getOrElse(false)
 
-  private def validClientId(payload: JsValue): Boolean =
-    (payload \ "clientId").toOption
-      .collect {
-        case x: JsString if clientIds.contains(x.value) => true
-      }
-      .getOrElse(false)
+  private def getClientName(payload: JsValue): Option[String] =
+    (payload \ "clientId").toOption.collect {
+      case x: JsString => clientIds.collectFirst { case (key, value) if value == x.value => key }
```

Comment:
clientIds no more a collection is int it ?

Reply:
it is a collection, it is a Map